### PR TITLE
add Dto, controller and mapper

### PR DESCRIPTION
**Pull request by OleksandrSaven**

### CODE REVIEW

File: src/main/java/com/app/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
```
Comment (ConfusedSatlan) [inline src/main/java/com/app/onlinebookstore/controller/BookController.java]: use /api/books

File: src/main/java/com/app/onlinebookstore/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+@Data
+public class CreateBookRequestDto {
+    private String title;
+    private String author;
+    private String isbn;
+    private BigDecimal price;
+    private String description;
+    private String coverImage;
+}
```
Comment (ConfusedSatlan) [inline src/main/java/com/app/onlinebookstore/dto/CreateBookRequestDto.java]: for such classes it's better to use record

File: src/main/java/com/app/onlinebookstore/mapper/BookMapper.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import com.app.onlinebookstore.model.Book;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper(config = MapperConfig.class)
+public interface BookMapper {
+    BookDto toDto(Book book);
+
+    @Mapping(target = "id", ignore = true)
+    Book toModel(CreateBookRequestDto requestDto);
```
Comment (ConfusedSatlan) [inline src/main/java/com/app/onlinebookstore/mapper/BookMapper.java]: could u explain me why u used @Mappiin(target = "id"..,, please???

**Review by ConfusedSatlan — APPROVED** [review]

**Review by OlegTuryk — APPROVED** [review]

**Review by vovasalyha — APPROVED** [review]
GJ

**Pull request merged**